src/Assignment4/Root/index.js

```diff
@@ -1,96 +1,23 @@
+import { useState } from "react";
+
 import Header from "../Header";
+import SearchComponent from "../SearchComponent";
 import CardComponent from "../CardComponent";
 import NitishPic from "../../logo/Nitish.jpg";
+import teamInformation from "../../common/data.json";
 
 const RootElement = () => {
-  const teamInformation = [
-    {
-      name: "Nitish Nivedan Barik",
-      company: "Accenture",
-      designaton: "Team Lead",
-      location: "Odisha",
-      id: "1",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Iswarya",
-      company: "Accenture",
-      designaton: "Software Engineer",
-      location: "bangalore",
-      id: "2",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Smriti",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "3",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Shaksham",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "4",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Shaksham",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "5",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Shekhawat",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "6",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Simran",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "7",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Sireesha",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "8",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Vamshi",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "9",
-      imageSource: NitishPic,
-    },
-    {
-      name: "Kalyan",
-      company: "Cognizant",
-      designaton: "Software Engineer",
-      location: "mumbai",
-      id: "9",
-      imageSource: NitishPic,
-    },
-  ];
+  const [currentCards, setCurrentCards] = useState(teamInformation);
   return (
     <>
       <Header />
-      {teamInformation &&
-        teamInformation.length > 0 &&
-        teamInformation.map((temmatesDetails) => (
+      <SearchComponent
+        setCurrentCards={setCurrentCards}
+        teamInformation={teamInformation}
+      />
+      {currentCards &&
```

Comment:
Will not it be more intuitive and readable to make a separate Cards component and pass the prop into it? Here you're passing the copy of currrentCards' state's single element to the CardComponent. Any theory or coding style behind it?

Reply:
didnt get the context. I already have a separate Card Component and passing props to it.  Any link or something can you share that will help me to understand.

Reply:
My bad. No need for Cards component.